prefect_email/credentials.py

```diff
@@ -90,8 +92,8 @@ class EmailServerCredentials(Block):
     _block_type_name = "Email Server Credentials"
     _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/3PcxFuO9XUqs7wU9MiUBMg/ca740e27815d15528373aced667f58b9/email__1_.png?h=250"  # noqa
 
-    username: str
-    password: SecretStr
+    username: Optional[str] = None
+    password: Optional[SecretStr] = ""
```

Comment:
I set it to "" because, although very rare, a SMTP password could be blank. So would None be treated differently than "", i.e. the calling code doesn't bother to set it? I could see None causing problems, though I didn't verify it.

Reply:
If we want to keep it as `""`, I think we should drop the type annotation `Optional` because [`Optional`](https://docs.python.org/3/library/typing.html#typing.Optional) indicates it can be None.

Reply:
See what I did to fix SecretStr - I realized it wasn't right while changing it.